driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -502,10 +502,11 @@ public ResponseHandler write(ResponseCallback callback) throws ConnectionExcepti
 
     public ResponseHandler write(ResponseCallback callback, boolean startTimeout) throws ConnectionException, BusyConnectionException {
 
-        Message.Request request = callback.request();
-
         ResponseHandler handler = new ResponseHandler(this, callback);
         dispatcher.add(handler);
+
+        // JAVA-1179: defensively guard against reusing same request object twice
+        Message.Request request = callback.request().copy();
```

Comment:
Is there any way around it or is an optimization to no copy not really worth it?